src/wallet/wallet.cpp

```diff
@@ -4486,4 +4486,25 @@ void CWallet::TopUpCallback(const std::set<CScript>& spks, ScriptPubKeyMan* spkm
     // Update scriptPubKey cache
     CacheNewScriptPubKeys(spks, spkm);
 }
+
+std::set<CExtPubKey> CWallet::GetActiveHDPubKeys() const
```

Comment:
I do agree on encapsulating this function but have to say that it's inconsistent to how previous commits on this PR were implemented (`gethdkeys` accesses the wallet internals to obtain the same information).

Yet, this is not blocking. Just a comment. We lack of an structure here.

Reply:
`gethdkeys` can get pubkeys for inactive descriptors, so it can't use this function.